### PR TITLE
keadm: unified access to version

### PR DESCRIPTION
**Pull request by zc2638**
**What type of PR is this?**

/kind feature

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

**Review by fisherxu — COMMENTED** [review]
For the release version, I think we can follow the way of kubeadm, as below:
1st. Get from remote, if the the remote Major version is bigger or if the Major versions are the same, but the remote Minor is bigger than client version, don't use the remote version.
2nd. Use the client version of keadm(git tag) to get the latest patch release version
3rd.  Use the client version of keadm(git tag)
4th. Use the version hard coded in keadm.

We can ensure keadm to get the latest patch release :) Maybe a little complex, but it's needed in the long term.

**Comment by fisherxu** [comment]
Ref: https://github.com/kubernetes/kubernetes/blob/master/cmd/kubeadm/app/util/version.go#L73

cc @zhu733756 @gy95 @TianTianBigWang

**Comment by zc2638** [comment]
PTAL @fisherxu

**Comment by gy95** [comment]
add some ut to help clear functions? 😂

**Comment by zc2638** [comment]
> add some ut to help clear functions? 😂

Done.
@gy95

File: keadm/cmd/keadm/app/cmd/util/common_test.go [hunk 1]
```diff
@@ -215,3 +215,165 @@ func TestComputeSHA512Checksum(t *testing.T) {
...
+		},
+		{
+			name:   "valid version with label and extra metadata",
+			input:  "v1.8.0-alpha.2.1231+afabd012389d53a.extra",
+			output: "v1.8.0-alpha.2",
+		},
+		{
+			name:   "valid patch version with label and extra metadata",
+			input:  "v1.11.3-beta.0.38+135cc4c1f47994",
+			output: "v1.11.2",
```
Comment (gy95) [inline keadm/cmd/keadm/app/cmd/util/common_test.go:241]: Is this by design? 1.11.3-beta to 1.11.2

File: keadm/cmd/keadm/app/cmd/util/common_test.go [hunk 2]
```diff
@@ -215,3 +215,165 @@ func TestComputeSHA512Checksum(t *testing.T) {
...
+		clientVersion string
+		output        string
+		expectedError bool
+	}
+	cases := []T{
+		{
+			name:          "valid: remote version is newer; return stable label [1]",
+			remoteVersion: "v1.12.0",
+			clientVersion: "v1.11.0",
+			output:        "stable-1.11",
```
Comment (gy95) [inline keadm/cmd/keadm/app/cmd/util/common_test.go:322]: is it ok for version with prefix `stable-`?

File: keadm/cmd/keadm/app/cmd/util/common_test.go (hunk 1 above)
Comment (zc2638) [reply keadm/cmd/keadm/app/cmd/util/common_test.go:241]: Yes, as @fisherxu said, directly borrowed from the design of k8s.

**Comment by zhu733756** [comment]
LGTM,  beta/init.go also needs a patch.

**Comment by zc2638** [comment]
> LGTM, beta/init.go also needs a patch.

`init` is special and will be fixed in the new pr.
@zhu733756

**Comment by TianTianBigWang** [comment]
LGTM
/lgtm

**Review by fisherxu — COMMENTED** [review]
/lgtm
/approve

**Pull request merged**